### ISSUE TITLE
DirectML + AMD GPU does not seem to work

### ISSUE THREAD

**Issue by rainbow-dolly** [royshil/obs-backgroundremoval#87]
I am using an updated Windows 11 system, OBS 27.2.3 and the Radeon Software 22.2.3 with AMD Radeon RX 6800 card.

When I switch to GPU (DirectML) in the plug-in (Version 0.4.0), nothing changes. My CPU still gets loaded at around 60% and no load goes to the GPU (I swap between CPU and GPU multiple times and nothing changes). Am I missing something to use this with my GPU, is my GPU not supported or is this a bug?

Thanks in advance. :)

**Comment by fabiomb** [comment]
i think the same happens with NVidia, i don't see a change in CPU use, is always over 70%

**Comment by ElSaico** [comment]
Same here, on Windows 10 with a Radeon 5600XT card. The OBS logs show this:

```
14:59:48.744: [obs-backgroundremoval] Exception during initialization: D:\3\s\onnxruntime\core\providers\dml\DmlExecutionProvider\src\FusedGraphKernel.cpp(64)\onnxruntime.dll!00007FF9502863BD: (caller: 00007FF950285978) Exception(2) tid(3974) 80004002 N�o h� suporte para esta interface
```

This error seems to have been reported on microsoft/onnxruntime#6457 also - perhaps it's not loading the provided DirectML?

**Comment by ubeogesh** [comment]
Same issue here on GeForce. Windows 10, RTX 2080 Ti.

I also get the "interface not supporter" error. 

`18:29:33.376: [obs-backgroundremoval] Exception during initialization: D:\3\s\onnxruntime\core\providers\dml\DmlExecutionProvider\src\FusedGraphKernel.cpp(64)\onnxruntime.dll!00007FFF340963BD: (caller: 00007FFF34095978) Exception(3) tid(34d4) 80004002 No such interface supported`

**Comment by royshil** [comment]
@ubeogesh can you try to zap the directml.dll from the plugin all together and see if it catches on the system32 one?
it might be the resolution here - to not include the dll in the plugin at all

**Comment by CraftingInC** [comment]
Confirmed, it does not work.

OBS : version 27.2.4
Plugin : obs-backgroundremoval-0.4.0-win64.zip

I'm using the pre-built binaries for 64-Bit Windows. I set it to GPU DirectML ( which is the only GPU choice it gives me ) and I am seeing that the GPU does NOT work. It uses the CPU no matter what choice I set it to. Also, It ramps up the CPU to 70% usage. Also, if I set the frame count to capture every 2 frames instead of just 1, then it lowers the CPU usage to 60%. But that is the only difference.

RAM : 64 Gigs
GPU: NVidia GTX 1650ti
CPU : i5-10300H
